persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/MetadataDAO.kt

```diff
@@ -0,0 +1,6 @@
+package com.wire.kalium.persistence.dao
+
+interface MetadataDAO {
+    suspend fun insertValue(value: String, key: String)
+    suspend fun valueByKey(key: String): String?
```

Comment:
Can we make it a `Flow` and use SQLDelight's coroutines extensions to do it? Or at least add another one function that returns a `Flow`.

This would enable us to make something like this:
```kotlin
    override suspend fun getSelfUser(): Flow<SelfUser> = 
        metadataDAO.valueByKey(SELF_USER_ID_KEY).filterNotNull().flatMapMerge { encodedValue ->
            val userId = Json.decodeFromString(encodedValue)
            userDAO.getUserByQualifiedID(userId)
                .filterNotNull()
                .map(userMapper::fromDaoModel)
        }
```

`getSelfUser` is not the best example, althought its code can be simplified.

But if we want to observe other `metadata` values, it could be a nice way of doing it.

Reply:
Yep sure, I can make it a flow didn't think it was useful here.